cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -357,12 +385,23 @@ impl ClientTrait for CatalogClient {
                         .collect::<Result<Vec<_>, _>>()?,
                 ))
             },
-            // I'm quite confident 10 can be stored as a NonZeroU32
-            unsafe { NonZeroU32::new_unchecked(10) },
+            RESPONSE_PAGE_SIZE,
         );
 
-        let results: Vec<SearchResult> = stream.try_collect().await?;
-        let count = Some(results.len() as u64);
+        let mut count: Option<u64> = None;
+        let mut results = Vec::new();
+        let mut stream = Box::pin(stream);
+
+        while let Some(result) = stream.next().await {
+            match result? {
+                StreamItem::TotalCount(tc) => {
+                    count = Some(tc);
+                },
+                StreamItem::Result(item) => {
+                    results.push(item);
+                },
+            }
+        }
```

Comment:
suggestion blocking: this isn't tested, and it's duplicated from above (except for limit, but that could be an `Option`).

This can be factored out into a helper function that just returns `count: u64, results: Vec<T>` which has two benefits:
1. We can test it
2. We only have to think about streams in one place

Like @dcarley said it makes me wonder about whether we should be mocking 1 level deeper. But I think we can still get around it for this change